tutor/src/components/course-listing/course.cjsx

```diff
@@ -0,0 +1,115 @@
+React = require 'react'
+classnames = require 'classnames'
+BS = require 'react-bootstrap'
+_ = require 'lodash'
+
+{Redirect, Link} = require 'react-router'
+
+Router = require '../../helpers/router'
+
+BRAND = 'Openstax'
+
+CourseBranding = React.createClass
+  displayName: 'CourseBranding'
+  propTypes:
+    isConceptCoach: React.PropTypes.bool.isRequired
+  render: ->
+    {isConceptCoach} = @props
+
+    if isConceptCoach
+      isBeta = false
+      brand = "#{BRAND} Concept Coach"
+    else
+      isBeta = true
+      brand = "#{BRAND} Tutor"
+
+    <p
+      className='course-listing-item-brand'
+      data-is-beta={isBeta}
+    >{brand}</p>
+
+Course = React.createClass
+  displayName: 'Course'
+
+  getCourseNameSegments: ->
+    {course, courseSubject} = @props
+
+    courseRegex = new RegExp(courseSubject, 'i')
+    courseNameMatches = courseRegex.exec(course.name)
+    if courseSubject and courseNameMatches
+      {index} = courseNameMatches
+
+      before = course.name.substring(0, index)
+      after = course.name.substring(index + courseSubject.length)
+
+      [before, courseSubject, after]
+    else
+      [course.name]
+
+  Controls: ->
+    {controls} = @props
+
+    <div className='course-listing-item-controls'>
+      {controls}
+    </div>
+
```

Comment:
Nice!  I puzzled over why you were running regex over the title, but it all become clear once I ran the code.  Is a clever approach to highlighting different parts of the course title.